src/plugins/hiddenColumns/test/editors.e2e.js

```diff
@@ -36,4 +36,39 @@ describe('HiddenColumns', () => {
       expect($mainHolder.scrollLeft()).toBe(startScrollLeft);
     });
   });
+
+  describe('Data change by typing with hiddenColumns', () => {
+    it('should correctly render the changed values subjected to validation when there is a hidden column next to it', async() => {
+      const hot = handsontable({
+        data: [[1, 2, 'Smith']],
+        colHeaders: true,
+        hiddenColumns: {
+          columns: [0], // hide the first column
+          indicators: true,
+          copyPasteEnabled: false
+        },
+        columns: [
+          {}, // the first empty column
+          { // the second numeric column
+            type: 'numeric',
+            allowInvalid: false,
+          },
+          {}, // the last column without validation
+        ]
+      });
+
+      hot.selectCell(0, 1);
+      keyDown('enter'); // open the editor
```

Comment:
I think `keyDownUp(...)` would be suited better to simulate a user's behaviour.

Reply:
Done by https://github.com/handsontable/handsontable/pull/7396/commits/741ea55cd3a3cc635ef4b7f68ee663f081f7a6d0.